scopus/abstract_retrieval.py

```diff
@@ -480,8 +485,16 @@ def __init__(self, EID, view='META_ABS', refresh=False):
             raise ValueError('view parameter must be one of ' +
                              ', '.join(allowed_views))
 
-        qfile = join(config.get('Directories', 'AbstractRetrieval'), EID)
-        url = "https://api.elsevier.com/content/abstract/eid/{}".format(EID)
+        if ID_type == 'auto':
+            ID_type = discover_id_type(EID)
+        else:
+            allowed_id_types = ('eid', 'pii', 'scopus_id', 'pubmed_id', 'doi')
+            if ID_type not in allowed_id_types:
+                raise ValueError('ID_type parameter must be one of ' +
+                                 ', '.join(allowed_id_types))
+
+        qfile = join(config.get('Directories', 'AbstractRetrieval'), EID.replace('/','_'))
```

Comment:
Still open question

Reply:
This is due to caching in function `scopus.utils.get_content`, line 85 which tries to create the file.

In Unix-based systems '/' is used as a path separator so it can't be used in file names. This is only a problem in the case of DOI, which always contains '/'.

Reply:
Though, I admit that replacement character '_' is kind-of arbitrarily chosen.

Reply:
Ah, I see. That's why in the search classes we cache files with the hashed-version of the filename. I think the underscore is okay.  Could you also update the note in the docstring, please?